### PR TITLE
[CEG-1029] Added new DCS sdk methods for trading, settlement, cranks

### PR DESCRIPTION
**Pull request by vizhang**

### CODE REVIEW

File: src/sdkV2.ts [hunk 1]
```diff
@@ -93,6 +93,29 @@ export default class CegaEvmSDKV2 {
...
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.bulkStartDCSTrades(vaultAddresses);
+  }
+
+  async submitDispute(vaultAddress: EvmAddress): Promise<ethers.providers.TransactionResponse> {
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.submitDispute(vaultAddress);
+  }
+
+  /**
```
Comment (vsai) [inline src/sdkV2.ts:115]: are the above methods trading methods too?

File: src/sdkV2.ts [hunk 2]
```diff
@@ -104,22 +127,133 @@ export default class CegaEvmSDKV2 {
...
-  async bulkStartTradesDcs(
+  async endAuctionDcs(
+    vaultAddress: EvmAddress,
+    auctionWinner: EvmAddress,
+    tradeStartDate: ethers.BigNumber,
+    aprBps: ethers.BigNumber,
+    oracleDataSource: OracleDataSourceDcs,
+    overrides: TxOverrides = {},
+  ): Promise<ethers.providers.TransactionResponse> {
+    if (!this._signer) {
```
Comment (vsai) [inline src/sdkV2.ts:150]: is `this._signer` being used anywhere in this function?
if not can remove this

File: src/sdkV2.ts [hunk 3]
```diff
@@ -104,22 +127,133 @@ export default class CegaEvmSDKV2 {
...
+    return cegaEntry.bulkRolloverDCSVaults(vaultAddresses, {
+      ...(await this._gasStation.getGasOraclePrices()),
+      ...overrides,
+    });
+  }
+
+  /**
+   * DISPUTE METHODS
+   */
+
```
Comment (vsai) [inline src/sdkV2.ts:208]: there was a dispute method above also. move that down here

File: src/sdkV2.ts [hunk 4]
```diff
@@ -93,7 +93,30 @@ export default class CegaEvmSDKV2 {
...
+    vaultAddresses: EvmAddress[],
+  ): Promise<ethers.providers.TransactionResponse> {
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.bulkStartDCSTrades(vaultAddresses);
+  }
+
+  async submitDispute(vaultAddress: EvmAddress): Promise<ethers.providers.TransactionResponse> {
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.submitDispute(vaultAddress);
+  }
```
Comment (consvic) [inline src/sdkV2.ts:113]: We already have these functions, think you need to update your branch to match main
https://github.com/cega-fi/cega-sdk-evm/blob/main/src/sdkV2.ts#L107-L125

**Review by consvic — DISMISSED** [review]

File: src/sdkV2.ts (hunk 4 above)
Comment (consvic) [reply src/sdkV2.ts:113]: also not sure if this is just github doing a weird diff

Comment (vizhang) [reply src/sdkV2.ts:113]: oh yeah it's github re-arranging the file (you can see that it removed it from lower on the page)

File: src/sdkV2.ts (hunk 1 above)
Comment (vizhang) [reply src/sdkV2.ts:100]: they're technically part of overall "trading flow" but are MM operations, i added their own comment section

File: src/sdkV2.ts (hunk 3 above)
Comment (vizhang) [reply src/sdkV2.ts:207]: moved

File: src/sdkV2.ts [hunk 5]
```diff
@@ -104,22 +112,150 @@ export default class CegaEvmSDKV2 {
...
 
+  async bulkSettleVaultsDcs(
+    vaultAddresses: EvmAddress[],
+  ): Promise<ethers.providers.TransactionResponse> {
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.bulkSettleDCSVaults(vaultAddresses);
+  }
+
+  /**
+   * SETTLEMENT METHODS
```
Comment (vsai) [inline src/sdkV2.ts:167]: CEGA SETTLEMENT METHODS

(because there's a "settle" method for the MMs to call also

**Review by vsai — APPROVED** [review]
one nit. otherwise lgtm

**Pull request merged**